lib/rules/no-unused-vars.js

```diff
@@ -584,6 +598,19 @@ module.exports = {
 
                     if (def) {
                         const type = def.type;
+                        const refUsedInArrayPatterns = variable.references && variable.references.some(ref => ref.identifier.parent.type === "ArrayPattern");
```

Comment:
```suggestion
                        const refUsedInArrayPatterns = variable.references.some(ref => ref.identifier.parent.type === "ArrayPattern");
```

`Variable#references` always exists. If the variable doesn't have references, it will be an empty array.